### ISSUE TITLE
http header Sec-Fetch-Dest: iframe instead of Sec-Fetch-Dest: document

### ISSUE THREAD

**Issue by guillaume-chervet** [cypress-io/cypress#20404]
### Current behavior

Fail because of HTTP headers:

Sec-Fetch-Dest: iframe

### Desired behavior

HTTP header:

Sec-Fetch-Dest: document

### Test code to reproduce

```js
describe('react-oidc', () => {
  
  beforeEach(() => {
    cy.visit('https://black-rock-0dc6b0d03.1.azurestaticapps.net')
  });

  it('displays two todo items by default', () => {
    cy.contains('Login').click()
    cy.wait(1100);
    cy.get("#Username").type("Bob");
    cy.wait(1100);
    cy.get("#Password").type("Bob").type('{enter}');

    cy.contains('Access Token');
  });

});
```

### Cypress Version

9.5.0

**Comment by colinmollenhour** [comment]
I'm trying to add a SameSite=Strict cookie to my app and it seems when the request is a GET the cookie is sent but when it is a POST with form fields it is not sent. In both cases the Sec-Fetch-Dest is "iframe" so I wonder if this has to do with that? That is, are SameSite=Strict cookies not sent on POST requests initiated from iframes?

![image](https://user-images.githubusercontent.com/38738/181137372-8a5dfbad-9c06-4f69-bc43-12a8dbde8a16.png)

**Comment by AtofStryker** [comment]
Hey @guillaume-chervet. Thank you for opening an issue. This should be available in `10.6.0` under the `experimentalModifyObstructiveThirdPartyCode` flag in #23720 to set the AUT iframe SEC destination metadata header to `document`. Can you enable the flag in `10.6.0` or greater and confirm it fixes your issue? I am going to close as I am fairly convinced this will work, but if it doesn't let me know and we can re open and investigate!
